solr/core/src/java/org/apache/solr/cloud/overseer/NodeMutator.java

```diff
@@ -76,7 +80,12 @@
       }
 
       if (needToUpdateCollection) {
```

Comment:
we no longer need this boolean, can check whether downedReplicas is empty or not to decide